src/client/content_mapblock.cpp

```diff
@@ -1272,6 +1275,12 @@ void MapblockMeshGenerator::drawRaillikeNode()
 	} else {
 		tile_index = rail_kinds[code].tile_index;
 		angle = rail_kinds[code].angle;
+
+		// Fix angle for fallback end texture
+		if (tile_index == end && angle >= 180 &&
+				f.tiledef[end].name == f.tiledef[straight].name) {
+			angle -= 180;
```

Comment:
Can't this be fixed ahead of time by rotating the texture in the nodedef? (we have a modified for that)